activejob/lib/active_job/queue_adapters.rb

```diff
@@ -48,5 +48,32 @@ module QueueAdapters
     autoload :SneakersAdapter
     autoload :SuckerPunchAdapter
     autoload :TestAdapter
+
+    ADAPTER = 'Adapter'.freeze
+
+    class << self
+      def instance(name, *adapter_args, &block)
+        hash = ([name, block] << adapter_args).hash
+
+        return instances.fetch(hash) if instances.key?(hash)
+
+        mutex.synchronize do
+          return instances.fetch(hash) if instances.key?(hash)
+
+          @instances[hash] ||= lookup(name).new(*adapter_args, &block)
```

Comment:
I think so - in particular, we may have an adapter that carries a connection object on it - we would not want to create multiple identical connections